tutorials/getting_started/another_model_example.md

```diff
@@ -0,0 +1,611 @@
+whatever structure you want, here; we chose this because it makes it clear what you have to write,
+and if you end up writing several different models or working on several tasks, this structure
+allows the repository to grow naturally.
+
+## Step three: write your DatasetReader
+
+Ok, now that our repository is set up, we'll start actually writing code ([link to the third
+commit](https://github.com/allenai/allennlp-as-a-library-example/commit/6be5e0b876325588803171a514f30da349542674)).
+The first thing we need is some data - what are we trying to build a model to predict?  In this
+example, we'll try to predict the "venue" of academic papers.  That is, given the title and
+abstract of a paper, we'll try to decide if it was (or should be) published in a "natural language
+processing" venue, a "machine learning" venue, or an "artificial intelligence" venue (all of those
+scare quotes are because this is a totally artificial task, and there aren't solid lines between
+these fields).
+
+We'll use the [open research corpus](labs.semanticscholar.org/corpus/) provided by the academic
+search engine [Semantic Scholar](http://semanticscholar.org), with a heuristically-edited "venue"
+field.  You can follow the link to see the full specification of this data, but it's provided as a
+JSON-lines file, where each JSON blob has at least these fields:
+
+```json
+{
+  "title": "A review of Web searching studies and a framework for future research",
+  "paperAbstract": "Research on Web searching is at an incipient stage. ...",
+  "venue": "{AI|ML|ACL}"
+}
+```
+
+Because we like writing tests (and you should too!), we'll write a test for our `DatasetReader` on
+some sample data before even writing any code.  We downloaded the data and took a sample of 10
+papers and made a little [test
+fixture](https://github.com/allenai/allennlp-as-a-library-example/blob/6be5e0b876325588803171a514f30da349542674/tests/fixtures/s2_papers.jsonl)
+out of them.  We'll use this fixture to make sure we can read the data as we expect.
+
+For our test, we inherit from `allennlp.common.testing.AllenNlpTestCase`.  For this simple test,
+that's not really necessary, but that base class does have some functionality around logging and
+cleaning up temporary files that can be nice for more complex tests.
+
+The basic API of the `DatasetReader` is just that we need to instantiate the object, then call
+`reader.read(data_file)`.  We do that in our test:
+
+```python
+from allennlp.common.testing import AllenNlpTestCase
+from my_library.dataset_readers import SemanticScholarDatasetReader
+
+class TestSemanticScholarDatasetReader(AllenNlpTestCase):
+    def test_read_from_file(self):
+        reader = SemanticScholarDatasetReader()
+        dataset = reader.read('tests/fixtures/s2_papers.jsonl')
+```
+
+Then we just want to make sure that the resulting dataset looks like we expect.  We'll refer you
+to the [dataset tutorial](TODO - how do I link this??) for a deeper dive on the `Dataset`,
```

Comment:
you can just link to the slug in the tutorials url. that tutorial is at http://allennlp.org/tutorials/data-pipeline

so you should be able to just do `[dataset tutorial](data-pipeline)`

Reply:
(that makes it work on the website but not in the github repo, unfortunately)

Reply:
Thanks.